src/api/hooks/useBestNumber.ts

```diff
@@ -5,17 +5,22 @@ import {BlockNumber} from '@polkadot/types/interfaces';
 import {useApi} from 'context/ChainApiContext';
 import useApiQuery from 'src/api/hooks/useApiQuery';
 
+const BEST_NUMBER_QUERY_KEY = 'api_derive_chain_bestNumber';
+
 export function useBestNumber(): BlockNumber | undefined {
   const {api} = useApi();
   const queryClient = useQueryClient();
-  const {data} = useApiQuery('api_derive_chain_bestNumber', (apiPromise) => apiPromise.derive.chain.bestNumber());
+  const {data} = useApiQuery(BEST_NUMBER_QUERY_KEY, (apiPromise) => apiPromise.derive.chain.bestNumber());
 
   useEffect(() => {
     let unsub: VoidFn | undefined;
 
     (async () => {
       unsub = await api?.derive.chain.bestNumber((blockNumber) => {
-        queryClient.setQueryData('api_derive_chain_bestNumber', blockNumber);
+        const currentBestNumber = queryClient.getQueryData<BlockNumber>(BEST_NUMBER_QUERY_KEY);
```

Comment:
I am trying to use that but it seems like if you call the setQueryData even with returning previous data it will cause rerenders. I will just keep it like this so the function wouldn't get called at all for now.